### PR TITLE
[CVE-2024-38808] [1.3] Bump spring-framework dependency to 2.9.13

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] New Roles/Permissions have a corresponding security dashboards plugin PR
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
this CVE fix requires Java 17 but 1.3 is set to Java 8

**Comment by cwperks** [comment]
This cannot be backported because 3.3.0 requires JDK >= 17

**Comment by DarshitChanpura** [comment]
bumped to 2.9.13 which is the latest spring-kafka version compatible with Java 8

File: build.gradle [hunk 1]
```diff
@@ -81,14 +81,14 @@ configurations.all {
         force 'commons-cli:commons-cli:1.3.1'
         force 'org.apache.httpcomponents:httpcore:4.4.12'
         force "org.apache.commons:commons-lang3:3.4"
-        force "org.springframework:spring-core:5.3.28"
-        force "org.springframework:spring-expression:5.3.28"
+        force "org.springframework:spring-core:5.3.39"
```
Comment (cwperks) [inline build.gradle:84]: Also force spring-context to 5.3.39 to resolve the whitesource error:

```
force "org.springframework:spring-context:5.3.39"
```

https://github.com/opensearch-project/security/pull/4947/checks?check_run_id=33937343697

Comment (DarshitChanpura) [reply build.gradle:84]: done

**Review by cwperks — APPROVED** [review]

**Pull request merged**